Update routes.go

ethanmuz opened the pull request:
I tried to give Plug some colors and fix the grammar a little bit

ethanmuz committed:
Update routes.go

I tried to give Plug some colors and fix the grammar a little bit

liam-middlebrook commented:
Thanks for taking the time to work on this, but I'm not really sure if I dig the clashing between the pink text and purple background. I would probably be more down to see the following scheme (although I'm not sure it would look good)

Purple Background
-> Pink Rectangle
  -> White Text
![2017-10-16-210006_362x292_scrot](https://user-images.githubusercontent.com/3920942/31646203-4ca33994-b2b5-11e7-84ff-c7a47af1ef4c.png)

Also if you're interested, making it so these pages are templated would be super useful. I have an example of what that may look like in the "pending_plugs" page that's referenced in the templates folder and inside of routes.go

ethanmuz committed:
Update routes.go

Tried to fix the colors to look more aesthetically pleasing

ethanmuz committed:
Update routes.go

This time added the alert script that reminds the uploader not to break the CoC, this was in the original code

ethanmuz commented:
Ok, I attempted to fix the colors a little bit, and I also centered the text. I am not sure exactly how to do templating though, but I will try to figure it out based on the template already in the Plug source code.

ethanmuz committed:
Plug HTML template

ethanmuz commented:
Ok, I just added my attempt at a template into the template folder.

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+<html>
+	<body style="background-image:url('https://i.imgur.com/YWfFxsh.png');" link="white" vlink="white" alink="white">
```
I'm not sure how I feel about a static image as the background. I think it would be neater if this was done with CSS. Also I'm not a fan of striped backgrounds.

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+			<a href="https://github.com/liam-middlebrook/csh-plug">Click here for CSH Plug on GitHub</a>
+		</div>
+		<script>
+ 		<!�� Put whatever script you want here ��>
```
Nitpick: If you're in JS you would use `// comment` or `/* comment */` for a comment.

Also please remove this script section.

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+		</p>
+		<div>
+		<p style="color: white";>
+                        Whatever form/image/other thing you want can go here
```
This is where you'd want to put some of the template-specific syntax, an example of how to do this can be found at https://github.com/gin-gonic/gin#html-rendering

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+<html>
+	<body style="background-image:url('https://i.imgur.com/YWfFxsh.png');" link="white" vlink="white" alink="white">
+                <center>
```
As much as I love `<center>` tags you should take the time to do this properly and avoid using deprecated tags. For more information on that please read https://developer.mozilla.org/en-US/docs/Web/HTML/Element/center

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+<html>
+	<body style="background-image:url('https://i.imgur.com/YWfFxsh.png');" link="white" vlink="white" alink="white">
+                <center>
+		<h1 style="color: white;">Title Goes Here!</h1>
```
Add this and the line below to use templated syntax.

liam-middlebrook commented on templates/plug_html_template.tmpl:
```diff
@@ -0,0 +1,23 @@
+		<h1 style="color: white;">Title Goes Here!</h1>
+		<h2 style="color: white;">Subtitle goes here!</h2>
+		<p style="color: white";>
+                        Informative text goes here
```
Add this to use templated syntax.

liam-middlebrook commented on routes.go:
```diff
@@ -97,14 +97,23 @@ func upload(c *gin.Context) {
-			<img src="`+S3PresignPlug(plug).String()+`"></img>
-		</div>
-	</body>
+		<body style="background-image:url('https://i.imgur.com/YWfFxsh.png');" link="white" vlink="white" alink="white">
```
You could move this over to a template, and just have the `S3PresignPlug(plug).String()` bit passed in as a variable to the template.

liam-middlebrook commented on routes.go:
```diff
@@ -117,24 +126,33 @@ func upload(c *gin.Context) {
-alert("The CSH CodeOfConduct Section 8 prohibits the sending of content that may harass others. Please review the CSH CodeOfConduct before uploading content to Plug.");
-		</script>
-	</body>
+		<body style="background-image:url('https://i.imgur.com/YWfFxsh.png');" link="white" vlink="white" alink="white">
```
Even though you wouldn't necessarily pass in any template variables to this, I'd recommend moving it over to a template file so it's easier to read through `routes.go`

liam-middlebrook reviewed (changes requested):
Good work, but there's still a bit to go before it'll be ready to get merged!

liam-middlebrook commented:
@ethanmuz also you'll want to rebase your changes on top of master. If you need help with that in person you should ask someone around floor like @sgreene570

liam-middlebrook commented:
I think most of the desired changes from here were covered by #7. If there's anything else you'd still like to add feel free to re-open this or create a new PR.
